components/Details.js

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from 'react';
+
+import { transformDetailsToFields } from './../utils/transformData';
+import { FormInput } from './Form/FormInput';
+export const Details = ({ detailsDataCallback, defaultValue }) => {
+  const [detailFields, setDetailFields] = useState(
+    transformDetailsToFields(defaultValue) || [],
+  );
+
+  const handleAddFields = (event) => {
+    event.preventDefault();
+    setDetailFields([...detailFields, { fieldName: '', fieldContent: '' }]);
+  };
+
+  const handleDeleteFields = (index, event) => {
+    event.preventDefault();
+    const data = [...detailFields];
+    data.splice(index, 1);
+    setDetailFields(data);
+  };
```

Comment:
```suggestion
  const handleDeleteFields = (indexToRemove, event) => {
    event.preventDefault();
    
    const newData = detailFields.filter((_, index) => index === indexToRemove);
    
    setDetailFields(newData);
  };
```

Reply:
Line 18: I think, that there should be a !==